.github/workflows/integration_test_dolphin.yml

```diff
@@ -394,7 +394,7 @@ jobs:
           aws-image-search-pattern: ${{ env.AWS_IMAGE_SEARCH_PATTERN }}
           aws-image-search-owners: ${{ env.AWS_IMAGE_SEARCH_OWNERS }}
   stop-node-builder-current:
-    if: ${{ always() }}
+    if: contains(github.event.pull_request.labels.*.name, 'A-dolphin' || github.ref == 'refs/heads/manta')
```

Comment:
I don't think this is safe, i thought of doing it this way before but then switched to always.
If someone labels the PR, the job runs. If they then unlabel the PR before CI is complete ( labelled by mistake ) this job will be skipped, leaving a zombie instance behind.

I think running always and ignoring success is the way to go here

Reply:
Actually that doesn't happen. Labeled and unlabeled and the job just finished the whole run.

Reply:
Also adding a label restarts both calamari and dolphin, if one of them was already present.

Reply:
I'll revert this change and leave it for a separate PR.